closed/src/java.base/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -102,15 +107,23 @@ protected void engineInit(Key key, SecureRandom random)
         if (ecKey instanceof ECPrivateKeyImpl keyImpl) {
             this.privateKey = keyImpl;
             this.publicKey = null;
-
+            Optional<ECOperations> opsOpt =
+                ECOperations.forParameters(this.privateKey.getParams());
```

Comment:
There are differences to this method relative to the head stream that I think are unwanted.

Reply:
Not sure what you mean here. Keep in mind that there is a change in the behaviour of the `ECDHKeyAgreement` for Java23 too. For example, keys are set to `null` before checking whether the key is a `PrivateKey`.

Reply:
I assume you're referring to
* [8320449: ECDHKeyAgreement should validate parameters before using them](https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/1f236d9598abe638550b7a61ce781f9da093e2a9)

Why would we not want to back-port that change here?

Reply:
Yeah, mostly. We'd want to back-port because, even thought there is no test that fails, the `ECDHKeyAgreement` class also check those parameters in Java22.